### DISCUSSION TITLE
Passing data from one component to another

### DISCUSSION THREAD

**Discussion by bxia-paul** [quasarframework/quasar#12437] [General - Components / Directives / etc] [answered]
I have a list that displays each post, however, I want that when you click on an item you are taken to a new page and the id of the post is in the URL. Below is the q-item where I want to set the attribute using the key.

`<q-item 
      v-for="post in pos" 
      :key="post.id" 
      to="/Details/${{post.id}}" 
      active-class="q-item-no-link-highlighting">`

This is the route:
{
    path: '/Details',
    component: () => import('layouts/PostDetailsLayout.vue'),
    children: [
      {
        path: '/Details/:id',component: () => import('pages/Details.vue')
        
      },
    ]
  }

What actually passes is not the integer post.id but the string post.id itself.
Any helpful suggestion?

**Comment by metalsadman** [comment, marked as answer]
use back ticks and syntax for template literal is incorrect, should be `${post.id}` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#syntax

```html
<q-item v-for="post in pos" :key="post.id" :to="`/Details/${post.id}`" active-class="q-item-no-link-highlighting">
```

**Reply by bxia-paul** [reply]
Thank you, I tried backticks but I didn't enclose it with "".
